io/eolearn/io/processing_api.py

```diff
@@ -283,43 +337,44 @@ def _request_payload(self, date_from, date_to, bbox, size_x, size_y):
     def _extract_data(self, eopatch, images, shape):
         """ Extract data from the received images and assign them to eopatch features
         """
+        rtypes = [rtype for rtype in self.request_types if rtype.id != 'bands']
+        itr_tifs = ((rtype, [img[rtype.id + '.tif'] for img in images], self.request_types[rtype]) for rtype in rtypes)
 
-        images = ((img['default.tif'], img['userdata.json']) for img in images)
-        images = [(img, meta.get('norm_factor', 0)) for img, meta in images]
+        for rtype, tifs, bands in itr_tifs:
+            for band in bands:
+                feature = rtype.feature_type, band
+                eopatch[feature] = self._extract_array(tifs, bands.index(band), shape, rtype.np_dtype)
 
-        for f_type, f_name_src, f_name_dst in self.additional_data:
-            eopatch[(f_type, f_name_dst)] = self._extract_additional_data(images, f_type, f_name_src, shape)
```

Comment:
The new changes don't support the writing of the additional data under the new name of the feature.

Reply:
Fixing it...

Reply:
Fixed.